package/base-files/files/lib/upgrade/nand.sh

```diff
@@ -229,22 +245,17 @@ nand_upgrade_ubinized() {
 	local ubi_file="$1"
 	local mtdnum="$(find_mtd_index "$CI_UBIPART")"
 
-	[ ! "$mtdnum" ] && {
-		CI_UBIPART="rootfs"
-		mtdnum="$(find_mtd_index "$CI_UBIPART")"
-	}
-
 	if [ ! "$mtdnum" ]; then
 		echo "cannot find mtd device $CI_UBIPART"
 		umount -a
 		reboot -f
 	fi
 
 	local mtddev="/dev/mtd${mtdnum}"
-	ubidetach -p "${mtddev}" || true
-	sync
+	ubidetach -p "${mtddev}" || :
```

Comment:
guess what's the only change? :) so either:
a) ubinized was already broken before these commits.
b) we are wrong and `sync` does something!
c) something obvious i'm not seeing and will make me look stupid soon.

i think we'll need a bit of your testing, sorry.

btw,

i copied you and changed true for :, thinking true was an external command... but they are both shell builtins! so what is the rationale for the change?

thanks!

Reply:
@dangowrt
FWIW, i can add some checks here or there for failure diagnostics in stuff not mine like ubinized, but in general that's not reliable. if i have to do shell, i always code with `set -e` and friends, and explicitly allow certain operations to fail if needed, and not the other way around.

in particular, i find this style horrible:
https://github.com/openwrt/openwrt/blob/827250481241f5c7706dfb5fd9cf07fdeb39f825/package/base-files/files/lib/upgrade/nand.sh#L186-L188
and that's my code... :(

it's probably too late to rewrite everything because this code is hard to test, we'll have to accept some silent failures.

the code cannot recover from a failed upgrade, it can't even properly respond to a browser with the failure message, so...

Reply:
Using `set -e` for this script is a good idea, I remember I wanted to do this years ago and started to prepare the code for that but then other things [tm] happened...

Reply:
> a) ubinized was already broken before these commits.

Retesting shows that. As I'm using blockd with autofs on UBIFS volume I was probably just waiting longer or haven't even triggered the volume to be mounted in first place and hence it was unmounted already...